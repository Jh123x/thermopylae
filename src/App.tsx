import { useState } from 'react';
import Terminal from './components/terminal';
import Quote from './components/quote/Quote';
import Settings from './components/settings/Settings';
import Particles from './components/particles/Particles';
import { useSettings } from './hooks';
import './app.css';

const App = () => {
    const [openSettings, setOpenSettings] = useState<boolean>(false);
    const { settings } = useSettings();
    return (
        <div className="main">
            <Particles />
            <div className="center">
                <img src={settings.backgroundImagePath} />
                <Terminal />
                <Quote />
            </div>
            <div className="top-right" onClick={() => setOpenSettings(true)}>
                <a><img alt="settings" src="./assets/gear.svg" width="20px" height="20px" /></a>
            </div>
            <div className={`settings-mask ${openSettings ? 'active' : ''}`}>
                {(<Settings onClose={() => setOpenSettings(false)} />)}
            </div>
        </div >
    );
}

export default App;
