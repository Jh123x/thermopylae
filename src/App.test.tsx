import { test, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import App from './App';


describe('App', () => {
    test('should render correctly', () => {
        const result = render(<App />);
        expect(result.baseElement).toMatchSnapshot();
    });
});
