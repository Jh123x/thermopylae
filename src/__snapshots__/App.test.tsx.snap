// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<body>
  <div>
    <div
      class="main"
    >
      <div
        id="particles-js"
        style="display: block;"
      />
      <img />
      <div
        class="center"
      >
        <div
          class="terminal"
        >
          <input
            list="bangs"
            type="text"
            value=""
          />
          <div
            id="bangs"
          />
        </div>
        <div
          class="quote"
          style="display: flex;"
        >
          <div
            class="text-line"
          >
            <span>
              Don't live with broken windows
            </span>
          </div>
          <div
            class="subtext-line"
          >
            <span>
              One broken window, left unrepaired for any substantial length of time, instills in the inhabitants of the building a sense of abandonmentâ€”a sense that the powers that be don't care about the building. So another window gets broken. People start littering. Graffiti appears. Serious structural damage begins. In a relatively short space of time, the building becomes damaged beyond the owner's desire to fix it, and the sense of abandonment becomes reality. 
 
 Don't leave "broken windows" (bad designs, wrong decisions, or poor code) unrepaired. Fix each one as soon as it is discovered. If there is insufficient time to fix it properly, then board it up. Perhaps you can comment out the offending code, or display a "Not Implemented" message, or substitute dummy data instead. Take some action to prevent further damage and to show that you're on top of the situation.
            </span>
          </div>
        </div>
      </div>
      <div
        class="top-right"
      >
        <a>
          <img
            alt="settings"
            height="20px"
            src="./assets/gear.svg"
            width="20px"
          />
        </a>
      </div>
      <div
        class="settings-mask "
      >
        <div
          class="settings"
        >
          <div
            class="close-button"
          >
            <img
              alt="cross"
              height="20px"
              src="/assets/cross.svg"
              width="20px"
            />
          </div>
          <div
            class="navbar"
          >
            <div
              class="menu active"
            >
              Background
            </div>
            <div
              class="menu "
            >
              Quote
            </div>
            <div
              class="menu "
            >
              Terminal
            </div>
          </div>
          <div
            class="divider"
          />
          <div
            class="settings-content"
          >
            <div
              class="background-settings"
            >
              <div
                class="input-group"
              >
                <label>
                  Background Image
                </label>
                <input
                  type="file"
                  value=""
                />
              </div>
              <div
                class="input-group"
              >
                <label>
                  Particles
                </label>
                <p>
                  <input
                    checked=""
                    type="checkbox"
                  />
                  <span>
                    Enabled
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
